app/Services/Interventions/MaintenanceContractService.php

```
@@ -0,0 +1,186 @@
+        $date = $date ?: now();
+
+        if ($contract->status !== MaintenanceContractStatus::ACTIVE) {
+            return false;
+        }
+
+        if (! $force && (is_null($contract->next_due_date) || $contract->next_due_date->gt($date))) {
+            return false;
+        }
+
+        return DB::transaction(function () use ($contract) {
+            $locked = MaintenanceContract::whereKey($contract->getKey())->lockForUpdate()->first();
```

PHPStan (2.2.7)

[error] Call to an undefined static method App\Models\Interventions\MaintenanceContract::whereKey().


(staticMethod.notFound)